src/Tests/BusinessLayer/TelemetryReportingTest.cpp

```diff
@@ -0,0 +1,959 @@
+
+                return false;
+            }
+
+            const unsigned char actualId = std::get<0>(input)[1];
+            bool res = actualId == expectedId_;
+
+            if (!res && listener->IsInterested())
+            {
+                ::std::ostream* os = listener->stream();
+                *os << "Actual id is " << actualId << std::endl;
+            }
+
+            return res;
+        }
+
+        virtual void DescribeTo(::std::ostream* os) const
+        {
+            *os << "Expected package ID = " << expectedId_ << std::endl;
+        }
+
+    private:
+        const unsigned char expectedId_;
+        const TelemetryReportingTest& containingTest_;
+    };
+
+    inline Matcher<std::tuple<const unsigned char*, int>> packageIdIs(unsigned char expectedId)
+    {
+        return MakeMatcher(new PackageIdMatcher(expectedId, *this));
+    }
+
+};
+
+/*
+ * Actual test cases are starting here
+ */
+
+/*
+ * This test tests for the correct structure of the sendKeyMotor package as defined in
+ * https://docs.google.com/spreadsheets/d/1soVLjeD9Sl7z7Z6cYMyn1fmn-cG7tx_pfFDsvgkCqMU/edit#gid=782574835
+ *
+ * The stuffing, framing and conversion is assumed to work correctly here. These methods are tested
+ * separately.
+ */
+TEST_F(TelemetryReportingTest, sendKeyMotorTest) // TODO create function which build the actual package to create more test cases in a easy way ...
+{
+    // prepare payload
+    const unsigned int expectedPackageLength = EXPECTED_PACKAGE_LENGTH_SEND_KEY_MOTOR;
+    const unsigned int payloadLength = expectedPackageLength - COBS_ADDITIONAL_FRAME_DATA_SIZE;
+    unsigned char data[payloadLength];
+    data[0] = CcsDefines::KEY_MOTOR_PKG_ID;
+    bool motor0Alive[] = {keyMotorData_->motor0Alive};
+    telemetryReporting_->writeBoolsIntoArray(data, 1, motor0Alive, 1);
+    telemetryReporting_->writeFloatIntoArray(data, 2, keyMotorData_->motor0SetCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 6, keyMotorData_->motor0SetVelocity);
+    telemetryReporting_->writeFloatIntoArray(data, 10, keyMotorData_->motor0BusCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 14, keyMotorData_->motor0BusVoltage);
+    telemetryReporting_->writeFloatIntoArray(data, 18, keyMotorData_->motor0VehicleVelocity);
+    bool motor1Alive[] = {keyMotorData_->motor1Alive};
+    telemetryReporting_->writeBoolsIntoArray(data, 22, motor1Alive, 1);
+    telemetryReporting_->writeFloatIntoArray(data, 23, keyMotorData_->motor1SetCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 27, keyMotorData_->motor1SetVelocity);
+    telemetryReporting_->writeFloatIntoArray(data, 31, keyMotorData_->motor1BusCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 35, keyMotorData_->motor1BusVoltage);
+    telemetryReporting_->writeFloatIntoArray(data, 39, keyMotorData_->motor1VehicleVelocity);
+    appendChecksum(data, payloadLength);
+    // do some additional data checks
+    ASSERT_THAT(data[0], Eq(0x01)); // packet id
```

Comment:
Yes, this is redundant now. I inserted these kind of assertions in the first place to ensure that I didn't introduce bugs during writing my test cases. With respect to that it makes sense that these are asserts rather than expects as the test case itself would be wrong.
It also checks (kind of) if the appendChecksum method affects the array at positions other than the last ones.